src/python/zigopt/suggestion/sampler/base.py

```diff
@@ -8,7 +8,7 @@
 
 
 class SuggestionSampler:
-  source = None
+  source: int
```

Comment:
looks a little odd, but still defaults to None (since types aren't actually used to coalesce to a default or anything?)?

Reply:
it will raise `AttributeError` if `source` isn't defined in the subclass. `SuggestionSampler` is an abstract class and all the subclasses must define this property.

Reply:
So the behavior could be different, but it's not in this case.

Reply:
Gotcha, cool 👍